packages/labs/analyzer/src/lib/model.ts

```diff
@@ -62,36 +89,50 @@ export class Module {
   }
 }
 
-export type Declaration = ClassDeclaration | VariableDeclaration;
+interface DeclarationInit {
+  name: string;
+}
 
-export interface VariableDeclarationInit {
+export abstract class Declaration {
   name: string;
+  constructor(init: DeclarationInit) {
+    this.name = init.name;
+  }
+  isVariableDeclaration(): this is VariableDeclaration {
```

Comment:
If we followed the module object pattern like TypeScript, we could also have a bunch of top-level functions for model object, like `analyzer.isClassDeclaration()`, etc.